src/plugins/dashboard/public/dashboard_listing/dashboard_unsaved_listing.tsx

```diff
@@ -156,7 +156,10 @@ export const DashboardUnsavedListing = ({
       const newItems = results.reduce((map, result) => {
         if (result.status === 'error') {
           hasError = true;
-          dashboardBackup.clearState(result.id);
+          if (result.error.statusCode === 404) {
+            // Save object not found error
+            dashboardBackup.clearState(result.id);
+          }
```

Comment:
This is ultimately what fixed this flakiness 🎉  We should **only** clear the state specifically if the CM service returned an "Object not found" error.